tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -592,6 +730,279 @@ class ScopedNormalizeDescriptor {
   SE_DISALLOW_COPY_AND_ASSIGN(ScopedNormalizeDescriptor);
 };
 
+// Turns a activation mode into a miopen activation mode descriptor with a scope around it
+class ScopedActivationDescriptor {
+public:
+  ScopedActivationDescriptor(ROCMExecutor* parent, dnn::ActivationMode activation_mode)
+    : parent_(parent)
+    , handle_(nullptr) {
+
+    miopenStatus_t status = wrap::miopenCreateActivationDescriptor(parent_, &handle_);
+    if (status != miopenStatusSuccess) {
+      
+      LOG(FATAL) << "call to miopenCreateActivationDescriptor failed: " << ToString(status);
+    }
+    else {
+      
+      miopenActivationMode_t miopen_activation_mode = miopenActivationPASTHRU;
+      double alpha = 0;
+      double beta = 0;
+      double gamma = 0;
+
+      switch (activation_mode) {
+
+      case dnn::ActivationMode::kNone:
+	miopen_activation_mode = miopenActivationPASTHRU;
+	break;
+
+      case dnn::ActivationMode::kSigmoid:
+	miopen_activation_mode = miopenActivationLOGISTIC;
+	break;
+
+      case dnn::ActivationMode::kRelu:
+	miopen_activation_mode = miopenActivationRELU;
+	break;
+
+      case dnn::ActivationMode::kRelu6:
+	miopen_activation_mode = miopenActivationRELU;
+	alpha = 6.0;
+	break;
+
+      case dnn::ActivationMode::kTanh:
+	miopen_activation_mode = miopenActivationTANH;
+	break;
+
```

Comment:
what about kBandPass?

Reply:
there does not seem to be a TF OP for it, dont know how to access its `X` attribute (which specifies the `[-X,X]` range, and hence left it out